iconic/UserProfile/views.py

```diff
@@ -19,6 +19,47 @@ def get(self, request):
         return Response(data, status=status.HTTP_200_OK)
 
 
+# A view for reading the number of followers
+class NumberOfFollowersView(APIView):
+    def get(self, request):
+        number_of_followers = Followers.objects.filter(user_id=request.user.id).count()
+        return Response({"user_id": request.user.id, "number_of_followers": number_of_followers},
+                        status=status.HTTP_200_OK)
+
+
+# A view for getting, adding and deleting followers
+class FollowersGetAddDeleteView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request):
+        user_id = request.data['user_id']
+        follower_id = request.user.id
+        current_follower = Followers.objects.get(user_id=user_id, follower=follower_id)
```

Comment:
If this user is not follower of user_id, than "Followers.objects.get(user_id=user_id, follower=follower_id)" line returns an error, and next code will not be executed